File: Programmers/2021-kakao-recruit/Lv1_신규 아이디 추천.py
import re

def solution(new_id):
    allow = '-_.'
    answer = ''
    # 1단계, 2단계
    for s in new_id:
        if s.isalpha():
            answer += s.lower()
        if s.isdigit():
            answer += s
        if s in allow:
            answer += s
    answer = re.sub('\.{2,}', '.', answer) # 3단계
    answer = re.sub('^\.|\.$', '', answer) # 4단계
    # 5단계
    if answer == '':
        answer = 'a'
    # 6단계
    if len(answer) >= 16:
        answer = answer[:15]
        answer = re.sub('\.$', '', answer)
    # 7단계
    if len(answer) <= 2:
        answer += answer[-1]*3
        answer = answer[:3]
    
    return answer

File: Programmers/2021-kakao-recruit/Lv1_신규아이디추천.js
function solution(new_id) {
  const id = new_id
    .toLowerCase()
    .replace(/[^\w-.]/g, "")
    .replace(/\.+/g, ".")
    .replace(/^\.|\.$/, "")
    .padEnd(1, "a")
    .slice(0, 15)
    .replace(/\.$/, "");

  return id.padEnd(3, id[id.length - 1]);
}
